File: api/Nathalie.Registry/Nathalie.Registry.DataLayer/Models/RegistryEntity.cs
﻿using System;
using System.Collections.Generic;
using MongoDB.Bson.Serialization.Attributes;
using MongoDB.Driver;

namespace Nathalie.Registry.DataLayer.Models
{
    [BsonIgnoreExtraElements]
    public class RegistryEntity
    {
        [BsonIgnore]
        public DateTime? RegistryDate { get; set; }

        public Template Template { get; set; }
        
        public bool IsFilledIn { get; set; }

        [BsonIgnoreIfNull]
        public IEnumerable<string> DataIds { get; set; }
        
        [BsonIgnore]
        public IEnumerable<RegistryEntityRow> Data { get; set; }
    }
}

File: app/nathalie-registry/src/app/models/registry-entity.ts
import {Template} from "./template";
import {RegistryEntityRow} from "./registry-entity-row";

export class RegistryEntity {
  public id: string;
  public template: Template;
  public registryDate: Date;
  public isFilledIn: boolean;
  public dataIds: Array<string>;
  public data: Array<RegistryEntityRow> = [];
}
